### PR TITLE
feat: Add Todo List App UI Mobile Design

### PR DESCRIPTION
**Pull request by moshoodabdullahi**
# This PR Adds the Todo List App UI Mobile Design following the [Todo List App Guide](https://github.com/bolah2009/command-line-guide/blob/master/guide/js/02-js-todo-app.md#2-mobile-design) and design layout of the assignment link of [TheOdinProject](https://www.theodinproject.com/lessons/node-path-javascript-todo-list) and [Front-End Mentor](https://www.frontendmentor.io/challenges/todo-app-Su1_KokOW)

This PR begins with the Todo List App's mobile view, uses dummy placeholders with Tailwind CSS, and makes some elements interactive by implementing hover effects.

| Frontend Mentor Design                          | Current Design                              |
| --------------------------------------- | ---------------------------------------------------- |
|image |image|

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -4,11 +4,199 @@
...
+            <label class="filters-list cursor-pointer hover:text-blue-400">
+              <input class="filters-input peer opacity-0" type="radio" name="filter" />
+              <span class="peer-checked:text-blue-600">Completed</span>
+            </label>
+          </div>
+        </div>
+        <div class="dragdrop fixed bottom-5 self-center text-base text-blue-300">
+          <p>Drag and drop to reorder list</p>
+        </div>
+      </div>
```
Comment (Yinkusadel) [inline index.html]: image
here is what you footer looks like please lets fix this

**Review by Yinkusadel — CHANGES_REQUESTED** [review]
fix that you should be good

File: index.html [hunk 2]
```diff
@@ -4,11 +4,199 @@
...
   </head>
 
-  <body class="m-0 flex min-h-screen min-w-80 place-items-center">
-    <main class="mx-auto my-0 max-w-7xl p-8 text-center" id="app" role="main"></main>
-    <script src="src/js/main.js" type="module"></script>
+  <body class="bg-blue-200 font-sans text-lg">
+    <main role="main">
+      <div class="relative m-auto flex w-96 flex-col justify-center">
+        <div
+          class="fixed left-0 right-0 top-0 -z-10 h-1/3 w-screen bg-mobile-light bg-cover bg-no-repeat"
```
Comment (Yinkusadel) [inline index.html]: we can use the shorthand property here

File: tailwind.config.cjs [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
-  content: [
-    "./index.html",
-    "./src/**/*.{js,ts,jsx,tsx}",
-  ],
+  content: ['./index.html', './src/**/*.{js,ts,jsx,tsx}'],
   theme: {
     extend: {},
+    fontFamily: {
+      sans: ['"Josefin Sans"', ...defaultTheme.fontFamily.sans],
+    },
```
Comment (Yinkusadel) [inline tailwind.config.cjs]: you should import the font directly to your html  file,

**Review by Yinkusadel — CHANGES_REQUESTED** [review]
please lets fix this too

File: index.html [hunk 4]
```diff
@@ -2,13 +2,238 @@
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
-    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <link rel="icon" href="public/icon-list.png" alt="list-icon" />
```
Comment (Yinkusadel) [inline index.html:5]: Is there a reason for this ???,if not lets remove it

File: index.html [hunk 5]
```diff
@@ -2,13 +2,238 @@
...
+        <form
+          class="list-item size-full h-14 list-none justify-between border border-blue-200 bg-blue-0 p-4 text-base text-blue-200"
+        >
+          <label class="relative flex cursor-pointer items-center rounded-t-lg font-sans">
+            <input
+              class="checkbox checked:bg-gradient-to-r peer mr-2.5 h-5 w-5 cursor-pointer appearance-none rounded-full border border-blue-200 from-blue-700 to-blue-800"
+              aria-label="input"
+              title="input-text"
+              type="checkbox"
+            />
```
Comment (Yinkusadel) [inline index.html]: this input does not have the checked mark ,also please remove the checkbox styled in the css whatever you want to do should be via tailwindconfig

image

File: index.html [hunk 6]
```diff
@@ -2,13 +2,238 @@
...
+            </label>
+            <label class="cursor-pointer hover:text-blue-400">
+              <input class="peer opacity-0" type="radio" name="filter" />
+              <span class="peer-checked:text-blue-600">Active</span>
+            </label>
+            <label class="cursor-pointer hover:text-blue-400">
+              <input class="peer opacity-0" type="radio" name="filter" />
+              <span class="peer-checked:text-blue-600">Completed</span>
+            </label>
+          </div>
```
Comment (Yinkusadel) [inline index.html]: lets use nav here instead,also i think the use of peer here is unnecessary please correct that also clicking your links makes it more of focused than hover or active

File: src/css/style.css [hunk 7]
```diff
@@ -1,3 +1,7 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
+.checkbox:checked {
+  background: linear-gradient(to right, hsl(192deg 100% 67%), hsl(280deg 87% 65%));
+}
```
Comment (Yinkusadel) [inline src/css/style.css]: we should remove this here and use the tailwind config

**Review by Yinkusadel — CHANGES_REQUESTED** [review]
All that and you should be good ,Nice work

File: index.html (hunk 4 above)
Comment (moshoodabdullahi) [reply index.html:5]: @Yinkusadel Thank for the Review, I am making use of this if you open the Todo List App on Netlify the The Front-End Mentor Logo has been replaced by that

File: src/css/style.css (hunk 7 above)
Comment (moshoodabdullahi) [reply src/css/style.css]: Done

File: index.html (hunk 5 above)
Comment (moshoodabdullahi) [reply index.html]: Done

**Comment by Yinkusadel** [comment]
> # This PR Adds the Todo List App UI Mobile Design following the [Todo List App Guide](https://github.com/bolah2009/command-line-guide/blob/master/guide/js/02-js-todo-app.md#2-mobile-design) and design layout of the assignment link of [TheOdinProject](https://www.theodinproject.com/lessons/node-path-javascript-todo-list) and [Front-End Mentor](https://www.frontendmentor.io/challenges/todo-app-Su1_KokOW)
> This PR begins with the Todo List App's mobile view, uses dummy placeholders with Tailwind CSS, and makes some elements interactive by implementing hover effects.
> 
> Frontend Mentor Design	Current Design
> image	image

Please update the current design also the PR description isn't good enough, and lets remove the **dummy placeholder**

File: index.html [hunk 8]
```diff
@@ -2,13 +2,238 @@
...
+            </label>
+            <label class="cursor-pointer hover:text-blue-400">
+              <input class="peer opacity-0" type="radio" name="filter" />
+              <span class="peer-checked:text-blue-600">Active</span>
+            </label>
+            <label class="cursor-pointer hover:text-blue-400">
+              <input class="peer opacity-0" type="radio" name="filter" />
+              <span class="peer-checked:text-blue-600">Completed</span>
+            </label>
+          </nav>
```
Comment (Yinkusadel) [inline index.html:247]: when using nav lists should follow ,those  input and span should as well be an anchor tag

also what i said about the hover effect and active state isn't fixed yet

File: tailwind.config.cjs [hunk 9]
```diff
@@ -1,10 +1,30 @@
...
+        400: 'hsl(235, 19%, 35%)',
+        500: 'hsl(234, 11%, 52%)',
+        600: 'hsl(220, 98%, 61%)',
+        700: 'hsl(192, 100%, 67%)',
+        800: 'hsl(280, 87%, 65%)',
+      },
+    },
+    backgroundImage: {
+      'mobile-light': "url('../../images/bg-mobile-light.jpg')",
+      'icon-checked': "url('../../images/icon-check.svg')",
```
Comment (Yinkusadel) [inline tailwind.config.cjs]: this doesn't seem to be well formatted

**Review by Yinkusadel — CHANGES_REQUESTED** [review]
Nice work

File: index.html [hunk 10]
```diff
@@ -2,13 +2,238 @@
...
+                >
+              </label>
+              <span class="remove h-3 w-3 cursor-pointer">
+                <img
+                  class="absolute right-7 top-72 h-4 w-4"
+                  alt="cross-icon"
+                  title="cross-icon-image"
+                  src="images/icon-cross.svg"
+                />
+              </span>
```
Comment (Yinkusadel) [inline index.html]: This should have an hover effect

File: index.html [hunk 11]
```diff
@@ -2,13 +2,238 @@
...
+              </label>
+            </li>
+          </ul>
+        </div>
+
+        <div
+          class="flex size-full h-14 justify-between rounded-b-lg border border-blue-200 bg-blue-0 p-4 text-base text-blue-500"
+        >
+          <div class="pt-1"><span>5</span> items left</div>
+          <div class="cursor-pointer">Clear Completed</div>
```
Comment (Yinkusadel) [inline index.html]: this should have a hover effect too and can as well be a button or link

**Review by Yinkusadel — CHANGES_REQUESTED** [review]
That's all from me

**Comment by moshoodabdullahi** [comment]
> > # This PR Adds the Todo List App UI Mobile Design following the [Todo List App Guide](https://github.com/bolah2009/command-line-guide/blob/master/guide/js/02-js-todo-app.md#2-mobile-design) and design layout of the assignment link of [TheOdinProject](https://www.theodinproject.com/lessons/node-path-javascript-todo-list) and [Front-End Mentor](https://www.frontendmentor.io/challenges/todo-app-Su1_KokOW)
> > This PR begins with the Todo List App's mobile view, uses dummy placeholders with Tailwind CSS, and makes some elements interactive by implementing hover effects.
> > Frontend Mentor Design	Current Design
> > image	image
> 
> Please update the current design also the PR description isn't good enough, and lets remove the **dummy placeholder**

File: index.html (hunk 8 above)
Comment (moshoodabdullahi) [reply index.html:247]: The label is the one wrapping everything and it has an hover effect

File: tailwind.config.cjs (hunk 9 above)
Comment (moshoodabdullahi) [reply tailwind.config.cjs]: Done

File: index.html (hunk 11 above)
Comment (moshoodabdullahi) [reply index.html]: Done

**Review by Yinkusadel — DISMISSED** [review]
nice

File: index.html [hunk 12]
```diff
@@ -2,13 +2,252 @@
...
   </head>
 
-  <body class="m-0 flex min-h-screen min-w-80 place-items-center">
-    <main class="mx-auto my-0 max-w-7xl p-8 text-center" id="app" role="main"></main>
-    <script src="src/js/main.js" type="module"></script>
+  <body class="bg-blue-200 text-lg">
+    <main role="main">
+      <section class="relative m-auto flex w-96 flex-col justify-center">
+        <div
+          class="fixed left-0 right-0 top-0 -z-10 h-1/3 w-screen bg-mobile-light bg-cover bg-no-repeat"
```
Comment (teckcipher) [inline index.html]: check tailwind to see if you can use shorthand for `left-0 right-0`. See https://tailwindcss.com/docs/top-right-bottom-left

File: index.html [hunk 13]
```diff
@@ -2,13 +2,252 @@
...
+        ></div>
+        <div class="my-28 mb-8 flex size-full justify-between">
+          <h1 class="text-4xl font-bold not-italic tracking-widest text-blue-100">TODO</h1>
+          <img class="h-8 w-8 cursor-pointer" src="images/icon-moon.svg" alt="theme-icon" />
```
Comment (teckcipher) [inline index.html]: We can use `size-` shorthand for `h-8 w-8`

File: index.html [hunk 14]
```diff
@@ -2,13 +2,252 @@
...
+        </div>
+
+        <form
+          class="list-item size-full h-14 list-none justify-between border border-blue-200 bg-blue-0 p-4 text-base text-blue-200"
+        >
+          <label class="relative flex cursor-pointer items-center rounded-t-lg font-sans">
+            <input
+              class="peer mr-2.5 h-5 w-5 cursor-pointer appearance-none rounded-full border border-blue-200 from-blue-700 to-blue-800 checked:border-none checked:bg-gradient-to-r"
```
Comment (teckcipher) [inline index.html]: same here, shorhand.

File: index.html [hunk 15]
```diff
@@ -2,13 +2,252 @@
...
+
+        <div>
+          <ul class="mt-5 w-96">
+            <li
+              class="flex size-full h-16 list-none items-center justify-between rounded-t-lg border border-blue-200 bg-blue-0 p-4 text-base text-blue-200"
+            >
+              <div class="flex max-w-[90%]">
+                <label class="relative flex cursor-pointer items-center pt-1.5">
+                  <input
+                    class="peer mr-2.5 h-5 w-5 cursor-pointer appearance-none rounded-full border border-blue-200 from-blue-700 to-blue-800 checked:border-none checked:bg-gradient-to-r checked:bg-center checked:bg-no-repeat"
```
Comment (teckcipher) [inline index.html]: .

File: index.html [hunk 16]
```diff
@@ -2,13 +2,252 @@
...
+              type="text"
+              placeholder="Create a new todo..."
+            />
+          </label>
+        </form>
+
+        <div>
+          <ul class="mt-5 w-96">
+            <li
+              class="flex size-full h-16 list-none items-center justify-between rounded-t-lg border border-blue-200 bg-blue-0 p-4 text-base text-blue-200"
```
Comment (teckcipher) [inline index.html]: Why use `size-full` when you will still use `h-16`? 🤔

File: index.html [hunk 17]
```diff
@@ -2,13 +2,252 @@
...
+                <img
+                  class="h-4 w-4 hover:scale-125"
+                  alt="cross-icon"
+                  title="cross-icon-image"
+                  src="images/icon-cross.svg"
+                />
+              </button>
+            </li>
+            <li
+              class="flex size-full h-16 list-none justify-between border border-blue-200 bg-blue-0 p-4 text-base text-blue-200"
```
Comment (teckcipher) [inline index.html]: same here

File: index.html [hunk 18]
```diff
@@ -2,13 +2,252 @@
...
+                  />
+                  <span
+                    class="font-bold text-blue-400 peer-checked:text-blue-500 peer-checked:line-through"
+                    >Complete online JavaScript course</span
+                  >
+                </label>
+              </div>
+              <button class="cursor-pointer">
+                <img
+                  class="h-4 w-4 hover:scale-125"
```
Comment (teckcipher) [inline index.html]: .

File: index.html [hunk 19]
```diff
@@ -2,13 +2,252 @@
...
+                  />
+                  <span
+                    class="font-bold text-blue-400 peer-checked:text-blue-500 peer-checked:line-through"
+                    >Pick up groceries</span
+                  >
+                </label>
+              </div>
+              <button class="cursor-pointer">
+                <img
+                  class="right-[22px] top-[410px] h-4 w-4 hover:scale-125"
```
Comment (teckcipher) [inline index.html]: All the `right` and `top` are still useful when the absolute is removed? 🤔

**Review by teckcipher — CHANGES_REQUESTED** [review]
make sure to apply suggestions to all occurrences

File: index.html (hunk 12 above)
Comment (moshoodabdullahi) [reply index.html]: Done

File: index.html (hunk 13 above)
Comment (moshoodabdullahi) [reply index.html]: Done

File: index.html (hunk 14 above)
Comment (moshoodabdullahi) [reply index.html]: Done

File: index.html (hunk 16 above)
Comment (moshoodabdullahi) [reply index.html]: Done

**Review by Yinkusadel — APPROVED** [review]

**Review by teckcipher — APPROVED** [review]

**Pull request merged**